app/javascript/components/users/user/forms/UpdateUserForm.jsx

```diff
@@ -15,36 +15,52 @@ import useUpdateUserForm from '../../../../hooks/forms/users/user/useUpdateUserF
 import PermissionChecker from '../../../../helpers/PermissionChecker';
 
 export default function UpdateUserForm({ user }) {
+  const { t } = useTranslation();
   const currentUser = useAuth();
-  const localesAPI = useLocales();
+
+  const canUpdateRole = PermissionChecker.hasManageUsers(currentUser);
+  const { data: roles } = useRoles({ enabled: canUpdateRole });
+  const { data: locales } = useLocales();
   const updateUserAPI = useUpdateUser(user?.id);
-  const { t } = useTranslation();
+
+  function currentLanguage() {
+    const language = user?.language;
+    const noDialect = language.substring(0, language.indexOf('-'));
+
+    if (locales?.[language] !== undefined) {
+      return language;
+    } if (noDialect !== '' && locales?.[noDialect] !== undefined) {
+      return noDialect;
+    }
+    return 'en';
+  }
```

Comment:
Shows the right language in the dropdown. Basically 
if users language is supported -> return that
elseif the users language is a dialect and we support the non-dialect version -> return that
fallback to english